Runtime/GameLiftCoreApi.cs

```diff
@@ -6,38 +6,42 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Amazon;
 using Amazon.GameLift;
-using AmazonGameLift.Editor;
 using AmazonGameLiftPlugin.Core;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement.Models;
+using AmazonGameLiftPlugin.Core.CredentialManagement;
+using AmazonGameLiftPlugin.Core.CredentialManagement.Models;
 using AmazonGameLiftPlugin.Core.Latency;
 using AmazonGameLiftPlugin.Core.Latency.Models;
 using AmazonGameLiftPlugin.Core.Shared;
+using AmazonGameLiftPlugin.Core.Shared.FileSystem;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement.Models;
-using UnityEngine;
 
 namespace AmazonGameLift.Runtime
 {
     public class GameLiftCoreApi
     {
+        public const string ConfigFilePath = "GameLiftSettings.yaml";
         private readonly GameLiftConfiguration _configuration;
         private readonly bool _isAnywhereMode;
-        private CoreApi _coreApi;
         private readonly IGameServerAdapter _gameServerAdapter;
 
         protected GameLiftCoreApi(GameLiftConfiguration configuration)
         {
             _configuration = configuration;
-            _coreApi = CoreApi.SharedInstance;
-            var fleetId = _coreApi.GetSetting("FleetId").Value;
-            var fleetLocation = _coreApi.GetSetting("FleetLocation").Value;
             if (_configuration.IsGameLiftAnywhere)
             {
+                var settings = new Settings<ClientSettingsKeys>(ConfigFilePath);
+                var fleetId = settings.GetSetting(ClientSettingsKeys.FleetId).Value;
+                var fleetLocation = settings.GetSetting(ClientSettingsKeys.FleetLocation).Value;
+                var credentials = new CredentialsStore(new FileWrapper());
                 var credentialsResponse =
-                    _coreApi.RetrieveAwsCredentials(_coreApi.GetSetting(SettingsKeys.CurrentProfileName).Value);
-                var gameLiftClient = new AmazonGameLiftClient(credentialsResponse.AccessKey, credentialsResponse.SecretKey);
+                    credentials.RetriveAwsCredentials(new RetriveAwsCredentialsRequest(){ProfileName = settings.GetSetting(ClientSettingsKeys.CurrentProfileName).Value});
```

Comment:
Are all of these settings assured to be set at this point? What if `settings.GetSetting` returns an Error? Is that something we need to consider?